smurfs/src/components/SmurfForm.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import axios from 'axios';
+class SmurfForm extends React.Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            id: Date.now,
+            name: '',
+            age: 0,
+            height: ''
+        };
+    }
+    handleChange = event => {
+        this.setState({ [event.target.name]: event.target.value });
+    }
+    handleSubmit = event => {
+        event.preventDefault();
+        console.log(this.state)
+        axios.post('http://localhost:3333/smurfs', this.state)
```

Comment:
So here you are doing the axios post request. This will successfully add a smurf to the backend server for you...but is it the best way? You are right that you want the smurf to be added WHEN you click the submit...but...no state is being updated this way. This is why you have to refresh the page to see the new smurf. Remember in your actions you are also doing an axios post request - however the action DOES update state, meaning your component will rerender. What if you dispatched the addSmurf action creator here?